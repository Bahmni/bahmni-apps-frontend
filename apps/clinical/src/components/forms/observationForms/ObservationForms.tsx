import {
  ComboBox,
  Tile,
  FormCard,
  FormCardContainer,
  SkeletonText,
} from '@bahmni-frontend/bahmni-design-system';
import { ObservationForm } from '@bahmni-frontend/bahmni-services';
import React, { useState, useMemo, useCallback } from 'react';
import { useTranslation } from 'react-i18next';
import { DEFAULT_FORM_API_NAMES } from '../../../constants/forms';
import useObservationFormsSearch from '../../../hooks/useObservationFormsSearch';
import styles from './styles/ObservationForms.module.scss';

interface ObservationFormsProps {
  onFormSelect?: (form: ObservationForm) => void;
  selectedForms?: ObservationForm[];
  onRemoveForm?: (formUuid: string) => void;
  // Pinned forms state passed from parent (required)
  pinnedForms: ObservationForm[];
  updatePinnedForms: (newPinnedForms: ObservationForm[]) => Promise<void>;
  isPinnedFormsLoading: boolean;
}

/**
 * ObservationForms component
 *
 * A component that displays a search interface for observation forms and a list of selected forms.
 * It allows users to search for observation forms by name, select them, and add them to the consultation.
 *
 * Features:
 * - ComboBox-based search interface
 * - Real-time filtering by form name
 * - Visual distinction for already-added forms
 * - Alphabetical ordering of search results
 * - Internationalization support
 * - Error handling and loading states
 */
const ObservationForms: React.FC<ObservationFormsProps> = React.memo(
  ({
    onFormSelect,
    selectedForms = [],
    onRemoveForm,
    pinnedForms,
    updatePinnedForms,
    isPinnedFormsLoading,
  }) => {
    const { t } = useTranslation();
    const [searchTerm, setSearchTerm] = useState('');

    const { forms: allForms, isLoading: isAllFormsLoading } =
      useObservationFormsSearch();

    const {
      forms: availableForms,
      isLoading: isSearchLoading,
      error: searchError,
    } = useObservationFormsSearch(searchTerm);

    // Validate and filter available forms - handle malformed data
    const validatedAvailableForms = useMemo(() => {
      return allForms.filter((form) => {
        // Check required properties
        if (!form || typeof form !== 'object') return false;
        if (!form.uuid || typeof form.uuid !== 'string') return false;
        if (!form.name || typeof form.name !== 'string') return false;
        return true;
      });
    }, [allForms]);

    // Use API names for filtering (these match the actual form names from backend)
    const defaultPinnedForms = validatedAvailableForms.filter((form) =>
      DEFAULT_FORM_API_NAMES.includes(form.name),
    );

    // Filter orphaned pinned forms - remove forms that are pinned but no longer available
    const validUserPinnedForms = useMemo(() => {
      return pinnedForms.filter((pinnedForm) => {
        return validatedAvailableForms.some(
          (availableForm) => availableForm.uuid === pinnedForm.uuid,
        );
      });
    }, [pinnedForms, validatedAvailableForms]);

    // Merge with user-pinned forms (avoid duplicates)
    const userPinnedUuids = validUserPinnedForms.map((f) => f.uuid);

    // Step 1: Get default forms that user hasn't pinned, sorted alphabetically
    const sortedDefaultForms = defaultPinnedForms
      .filter((f) => !userPinnedUuids.includes(f.uuid))
      .sort((a, b) => a.name.localeCompare(b.name));

    // Step 2: Get user-pinned forms, sorted alphabetically
    const sortedUserPinnedForms = [...validUserPinnedForms].sort((a, b) =>
      a.name.localeCompare(b.name),
    );

    // Step 3: Combine - defaults first, then user-pinned
    const allPinnedForms = [...sortedDefaultForms, ...sortedUserPinnedForms];

    const handleSearch = useCallback((searchQuery: string) => {
      setSearchTerm(searchQuery);
    }, []);

    const handleOnChange = useCallback(
      (data: {
        selectedItem?: { id: string; label: string; disabled?: boolean } | null;
      }) => {
        const selectedItem = data.selectedItem;
        if (!selectedItem?.id || selectedItem.disabled) {
          return;
        }
        const form = availableForms.find(
          (f: ObservationForm) => f.uuid === selectedItem.id,
        );
        if (form) {
          onFormSelect?.(form);
        }
      },
      [availableForms, onFormSelect],
    );

    const searchResults = useMemo(() => {
      if (isSearchLoading) {
        return [
          {
            id: '',
            label: t('OBSERVATION_FORMS_LOADING_FORMS'),
            disabled: true,
          },
        ];
      }

      if (searchError) {
        return [
          {
            id: '',
            label: t('OBSERVATION_FORMS_ERROR_LOADING_FORMS'),
            disabled: true,
          },
        ];
      }

      if (availableForms.length === 0 && searchTerm.length > 0) {
        return [
          {
            id: '',
            label: t('OBSERVATION_FORMS_NO_FORMS_FOUND'),
            disabled: true,
          },
        ];
      }

      if (availableForms.length === 0) {
        return [
          {
            id: '',
            label: 'No forms available',
            disabled: true,
          },
        ];
      }

      // Map forms to ComboBox items with proper labeling for already selected forms
      const results = availableForms.map((form: ObservationForm) => {
        const isAlreadySelected = selectedForms.some(
          (selected: ObservationForm) => selected.uuid === form.uuid,
        );

        return {
          id: form.uuid,
          label: isAlreadySelected
            ? `${form.name} (${t('OBSERVATION_FORMS_FORM_ALREADY_ADDED')})`
            : form.name,
          disabled: isAlreadySelected,
        };
      });

      return results;
    }, [
      isSearchLoading,
      searchError,
      searchTerm,
      availableForms,
      selectedForms,
      t,
    ]);

    return (
      <Tile
        className={styles.observationFormsTile}
        data-testid="observation-forms-tile"
      >
        <div
          className={styles.observationFormsTitle}
          data-testid="observation-forms-title"
        >
          {t('OBSERVATION_FORMS_SECTION_TITLE')}
        </div>

        <div data-testid="observation-forms-search-section">
          <ComboBox
            id="observation-forms-search"
            placeholder={t('OBSERVATION_FORMS_SEARCH_PLACEHOLDER')}
            items={searchResults}
            itemToString={(item) => item?.label ?? ''}
            onChange={handleOnChange}
            onInputChange={handleSearch}
            size="md"
            autoAlign
            disabled={isSearchLoading}
            aria-label={t('OBSERVATION_FORMS_SEARCH_ARIA_LABEL')}
            data-testid="observation-forms-search-combobox"
          />
        </div>

        {selectedForms && selectedForms.length > 0 && (
          <div data-testid="added-forms-section">
            <FormCardContainer
              title={t('OBSERVATION_FORMS_ADDED_FORMS')}
              dataTestId="added-forms-container"
            >
              {selectedForms.map((form: ObservationForm) => (
                <FormCard
                  key={form.uuid}
                  title={form.name}
                  icon="fa-file-lines"
                  actionIcon="fa-times"
                  onOpen={() => onFormSelect?.(form)}
                  onActionClick={() => onRemoveForm?.(form.uuid)}
                  dataTestId={`selected-form-${form.uuid}`}
                  ariaLabel={`Open ${form.name} form`}
                />
              ))}
            </FormCardContainer>
          </div>
        )}

        <div data-testid="pinned-forms-section">
          <FormCardContainer
            title={t('DEFAULT_AND_PINNED_FORMS_TITLE')}
            showNoFormsMessage={
              !isAllFormsLoading &&
              allPinnedForms.length === 0 &&
              defaultPinnedForms.length === 0
            }
            noFormsMessage={t('DEFAULT_AND_PINNED_FORMS_NO_FORMS_FOUND')}
            dataTestId="pinned-forms-container"
          >
            {isAllFormsLoading || isPinnedFormsLoading ? (
              <SkeletonText
                width="100%"
                lineCount={3}
                testId="pinned-forms-skeleton"
              />
            ) : (
              allPinnedForms.map((form: ObservationForm) => (
                <FormCard
                  key={form.uuid}
                  title={form.name}
                  icon="fa-file-lines"
                  actionIcon={
                    !DEFAULT_FORM_API_NAMES.includes(form.name)
                      ? 'fa-thumbtack'
                      : undefined
                  }
                  onOpen={() => onFormSelect?.(form)}
                  onActionClick={() => {
                    const newPinnedForms = pinnedForms.filter(
                      (f) => f.uuid !== form.uuid,
                    );
                    updatePinnedForms(newPinnedForms);
                  }}
                  dataTestId={`pinned-form-${form.uuid}`}
                  ariaLabel={`Open ${form.name} form`}
                />
              ))
            )}
          </FormCardContainer>
        </div>
      </Tile>
    );
  },
);

ObservationForms.displayName = 'ObservationForms';

export default ObservationForms;
