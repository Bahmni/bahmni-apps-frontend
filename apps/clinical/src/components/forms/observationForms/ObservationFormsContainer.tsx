import {
  ActionArea,
  Icon,
  ICON_SIZE,
} from '@bahmni-frontend/bahmni-design-system';
import { ObservationForm } from '@bahmni-frontend/bahmni-services';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { DEFAULT_FORM_API_NAMES } from '../../../constants/forms';
import styles from './styles/ObservationFormsContainer.module.scss';

interface ObservationFormsContainerProps {
  // Callback to notify parent when form viewing starts/ends
  onViewingFormChange: (viewingForm: ObservationForm | null) => void;
  // The currently viewing form (passed from parent)
  viewingForm?: ObservationForm | null;
  // Callback to remove form from selected forms list
  onRemoveForm?: (formUuid: string) => void;
  // Pinned forms state passed from parent (required)
  pinnedForms: ObservationForm[];
  updatePinnedForms: (newPinnedForms: ObservationForm[]) => Promise<void>;
}

/**
 * ObservationFormsWrapper component
 *
 * Wraps the ObservationForms component with additional functionality that was extracted from ConsultationPad.
 * This component manages its own state for selected forms and viewing form,
 * and renders its own ActionArea when viewing a form.
 *
 * When viewing a form, it takes over the entire UI with its own ActionArea.
 * When not viewing a form, it renders just the observation forms component.
 */
const ObservationFormsContainer: React.FC<ObservationFormsContainerProps> = ({
  onViewingFormChange,
  viewingForm: externalViewingForm,
  onRemoveForm,
  pinnedForms,
  updatePinnedForms,
}) => {
  const { t } = useTranslation();

  // Use the external viewingForm from parent
  const viewingForm = externalViewingForm;

  // Check if current form is pinned
  const isCurrentFormPinned = viewingForm
    ? pinnedForms.some((form) => form.uuid === viewingForm.uuid)
    : false;

  const handlePinToggle = (e: React.MouseEvent) => {
    e.preventDefault();
    e.stopPropagation();
    if (viewingForm) {
      let newPinnedForms;
      if (isCurrentFormPinned) {
        newPinnedForms = pinnedForms.filter(
          (form) => form.uuid !== viewingForm.uuid,
        );
      } else {
        newPinnedForms = [...pinnedForms, viewingForm];
      }
      updatePinnedForms(newPinnedForms);
    }
  };

  const handleDiscardForm = () => {
    // Remove the form from selected forms list if callback is provided
    if (viewingForm && onRemoveForm) {
      onRemoveForm(viewingForm.uuid);
    }
    // Close the form view
    onViewingFormChange(null);
  };

  const handleSaveForm = () => {
    // TODO: Implement form saving logic
    onViewingFormChange(null);
  };

  // Form view content when a form is selected
  const formViewContent = (
    <div className={styles.formView}>
      <div className={styles.formContent}>
        {/* TODO: Actual form rendering will be implemented here */}
        {/* For now, show empty content as form rendering is not yet implemented */}
      </div>
    </div>
  );

  // Create a custom title with pin icon
  const formTitleWithPin = (
    <div className={styles.formTitleContainer}>
      <span>{viewingForm?.name}</span>
      {!DEFAULT_FORM_API_NAMES.includes(viewingForm?.name ?? '') && (
        <div
          onClick={handlePinToggle}
          className={`${styles.pinIconContainer} ${isCurrentFormPinned ? styles.pinned : styles.unpinned}`}
          title={isCurrentFormPinned ? 'Unpin form' : 'Pin form'}
        >
          <Icon id="pin-icon" name="fa-thumbtack" size={ICON_SIZE.SM} />
        </div>
      )}
    </div>
  );

  // If viewing a form, render the form with its own ActionArea
  if (viewingForm) {
    return (
      <ActionArea
        className={styles.formViewActionArea}
        title={formTitleWithPin as unknown as string}
        primaryButtonText={t('OBSERVATION_FORM_SAVE_BUTTON')}
        onPrimaryButtonClick={handleSaveForm}
        isPrimaryButtonDisabled={false}
        secondaryButtonText={t('OBSERVATION_FORM_DISCARD_BUTTON')}
        onSecondaryButtonClick={handleDiscardForm}
        tertiaryButtonText={t('OBSERVATION_FORM_BACK_BUTTON')}
        onTertiaryButtonClick={() => {
          onViewingFormChange(null);
        }}
        content={formViewContent}
      />
    );
  }

  // If no form is being viewed, render nothing
  return null;
};

export default ObservationFormsContainer;
