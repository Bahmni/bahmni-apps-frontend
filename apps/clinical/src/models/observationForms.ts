/**
 * User properties for pinned forms service
 */
export interface UserProperties {
  defaultLocale?: string;
  favouriteObsTemplates?: string;
  pinnedObsTemplates?: string;
  favouriteWards?: string;
  loginAttempts?: string;
  recentlyViewedPatients?: string;
  [key: string]: unknown;
}

/**
 * User data for pinned forms service
 */
export interface UserData {
  uuid: string;
  username: string;
  userProperties?: UserProperties;
  [key: string]: unknown;
}
