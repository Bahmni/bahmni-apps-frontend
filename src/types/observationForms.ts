export interface ObservationForm {
  uuid: string;
  name: string;
  version?: string;
  formName?: string;
  formUuid?: string;
}
