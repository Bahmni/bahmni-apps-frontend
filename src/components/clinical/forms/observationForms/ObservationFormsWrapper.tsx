import React from 'react';
import { useTranslation } from 'react-i18next';
import ActionArea from '@components/common/actionArea/ActionArea';
import { ObservationForm } from '@types/observationForms';
import * as styles from '../../consultationPad/styles/ConsultationPad.module.scss';

interface ObservationFormsWrapperProps {
  // Callback to notify parent when form viewing starts/ends
  onViewingFormChange: (viewingForm: ObservationForm | null) => void;
  // The currently viewing form (passed from parent)
  viewingForm?: ObservationForm | null;
  // Callback to remove form from selected forms list
  onRemoveForm?: (formUuid: string) => void;
}

/**
 * ObservationFormsWrapper component
 *
 * Wraps the ObservationForms component with additional functionality that was extracted from ConsultationPad.
 * This component manages its own state for selected forms and viewing form,
 * and renders its own ActionArea when viewing a form.
 *
 * When viewing a form, it takes over the entire UI with its own ActionArea.
 * When not viewing a form, it renders just the observation forms component.
 */
const ObservationFormsWrapper: React.FC<ObservationFormsWrapperProps> = ({
  onViewingFormChange,
  viewingForm: externalViewingForm,
  onRemoveForm,
}) => {
  const { t } = useTranslation();

  // Use the external viewingForm from parent
  const viewingForm = externalViewingForm;

  const handleDiscardForm = () => {
    // Remove the form from selected forms list if callback is provided
    if (viewingForm && onRemoveForm) {
      onRemoveForm(viewingForm.uuid);
    }
    // Close the form view
    onViewingFormChange(null);
  };

  const handleSaveForm = () => {
    // TODO: Implement form saving logic
    onViewingFormChange(null);
  };

  // Form view content when a form is selected
  const formViewContent = (
    <div className={styles.formView}>
      <div className={styles.formContent}>
        {/* TODO: Actual form rendering will be implemented here */}
        {/* For now, show empty content as form rendering is not yet implemented */}
      </div>
    </div>
  );

  // If viewing a form, render the form with its own ActionArea
  if (viewingForm) {
    return (
      <ActionArea
        className={styles.formViewActionArea}
        title={viewingForm.name}
        primaryButtonText={t('OBSERVATION_FORM_SAVE_BUTTON')}
        onPrimaryButtonClick={handleSaveForm}
        isPrimaryButtonDisabled={false}
        secondaryButtonText={t('OBSERVATION_FORM_DISCARD_BUTTON')}
        onSecondaryButtonClick={handleDiscardForm}
        tertiaryButtonText={t('OBSERVATION_FORM_BACK_BUTTON')}
        onTertiaryButtonClick={() => {
          onViewingFormChange(null);
        }}
        content={formViewContent}
      />
    );
  }

  // If no form is being viewed, render nothing
  return null;
};

export default ObservationFormsWrapper;
