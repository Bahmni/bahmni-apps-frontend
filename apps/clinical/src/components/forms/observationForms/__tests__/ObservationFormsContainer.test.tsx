import { render, screen, fireEvent } from '@testing-library/react';
import React from 'react';
import { ObservationForm } from '@types/observationForms';
import ObservationFormsContainer from '../ObservationFormsContainer';

// Mock the translation hook
jest.mock('react-i18next', () => ({
  useTranslation: jest.fn(() => ({
    t: jest.fn((key) => `translated_${key}`),
  })),
}));

// Mock ActionArea component
jest.mock('@components/common/actionArea/ActionArea', () => {
  return jest.fn(
    ({
      className,
      title,
      primaryButtonText,
      onPrimaryButtonClick,
      isPrimaryButtonDisabled,
      secondaryButtonText,
      onSecondaryButtonClick,
      tertiaryButtonText,
      onTertiaryButtonClick,
      content,
    }) => (
      <div data-testid="action-area" className={className}>
        <div data-testid="action-area-title">{title}</div>
        <div data-testid="action-area-content">{content}</div>
        <div data-testid="action-area-buttons">
          <button
            data-testid="primary-button"
            disabled={isPrimaryButtonDisabled}
            onClick={onPrimaryButtonClick}
          >
            {primaryButtonText}
          </button>
          <button
            data-testid="secondary-button"
            onClick={onSecondaryButtonClick}
          >
            {secondaryButtonText}
          </button>
          <button data-testid="tertiary-button" onClick={onTertiaryButtonClick}>
            {tertiaryButtonText}
          </button>
        </div>
      </div>
    ),
  );
});

// Mock styles
jest.mock('../../consultationPad/styles/ConsultationPad.module.scss', () => ({
  formView: 'formView',
  formContent: 'formContent',
  formViewActionArea: 'formViewActionArea',
}));

describe('ObservationFormsContainer', () => {
  const mockForm: ObservationForm = {
    name: 'Test Form',
    uuid: 'test-form-uuid',
    id: 1,
    privileges: [],
  };

  const defaultProps = {
    onViewingFormChange: jest.fn(),
    viewingForm: null,
    onRemoveForm: jest.fn(),
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('Rendering and Structure', () => {
    it('should render ActionArea when viewingForm is provided', () => {
      render(
        <ObservationFormsContainer {...defaultProps} viewingForm={mockForm} />,
      );

      expect(screen.getByTestId('action-area')).toBeInTheDocument();
      expect(screen.getByTestId('action-area-title')).toHaveTextContent(
        'Test Form',
      );
    });

    it('should match the snapshot when viewing a form', () => {
      const { container } = render(
        <ObservationFormsContainer {...defaultProps} viewingForm={mockForm} />,
      );
      expect(container).toMatchSnapshot();
    });

    it('should match the snapshot when not viewing a form', () => {
      const { container } = render(
        <ObservationFormsContainer {...defaultProps} viewingForm={null} />,
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe('ActionArea Configuration', () => {
    it('should configure ActionArea with correct props', () => {
      render(
        <ObservationFormsContainer {...defaultProps} viewingForm={mockForm} />,
      );

      const actionArea = screen.getByTestId('action-area');
      expect(actionArea).toHaveClass('formViewActionArea');

      expect(screen.getByTestId('primary-button')).toHaveTextContent(
        'translated_OBSERVATION_FORM_SAVE_BUTTON',
      );
      expect(screen.getByTestId('secondary-button')).toHaveTextContent(
        'translated_OBSERVATION_FORM_DISCARD_BUTTON',
      );
      expect(screen.getByTestId('tertiary-button')).toHaveTextContent(
        'translated_OBSERVATION_FORM_BACK_BUTTON',
      );
    });
  });

  describe('Button Click Handlers', () => {
    it('should call onViewingFormChange with null when Save button is clicked', () => {
      const mockOnViewingFormChange = jest.fn();
      render(
        <ObservationFormsContainer
          {...defaultProps}
          onViewingFormChange={mockOnViewingFormChange}
          viewingForm={mockForm}
        />,
      );

      const saveButton = screen.getByTestId('primary-button');
      fireEvent.click(saveButton);

      expect(mockOnViewingFormChange).toHaveBeenCalledWith(null);
    });

    it('should call onViewingFormChange with null when Back button is clicked', () => {
      const mockOnViewingFormChange = jest.fn();
      render(
        <ObservationFormsContainer
          {...defaultProps}
          onViewingFormChange={mockOnViewingFormChange}
          viewingForm={mockForm}
        />,
      );

      const backButton = screen.getByTestId('tertiary-button');
      fireEvent.click(backButton);

      expect(mockOnViewingFormChange).toHaveBeenCalledWith(null);
    });

    it('should call both onRemoveForm and onViewingFormChange when Discard button is clicked', () => {
      const mockOnViewingFormChange = jest.fn();
      const mockOnRemoveForm = jest.fn();
      render(
        <ObservationFormsContainer
          {...defaultProps}
          onViewingFormChange={mockOnViewingFormChange}
          onRemoveForm={mockOnRemoveForm}
          viewingForm={mockForm}
        />,
      );

      const discardButton = screen.getByTestId('secondary-button');
      fireEvent.click(discardButton);

      expect(mockOnRemoveForm).toHaveBeenCalledWith('test-form-uuid');
      expect(mockOnViewingFormChange).toHaveBeenCalledWith(null);
    });

    it('should only call onViewingFormChange when Discard button is clicked and onRemoveForm is not provided', () => {
      const mockOnViewingFormChange = jest.fn();
      render(
        <ObservationFormsContainer
          {...defaultProps}
          onViewingFormChange={mockOnViewingFormChange}
          onRemoveForm={undefined}
          viewingForm={mockForm}
        />,
      );

      const discardButton = screen.getByTestId('secondary-button');
      fireEvent.click(discardButton);

      expect(mockOnViewingFormChange).toHaveBeenCalledWith(null);
    });
  });

  describe('Form Display', () => {
    it('should display the correct form name in the title', () => {
      const customForm: ObservationForm = {
        name: 'Custom Form Name',
        uuid: 'custom-uuid',
        id: 2,
        privileges: [],
      };

      render(
        <ObservationFormsContainer
          {...defaultProps}
          viewingForm={customForm}
        />,
      );

      expect(screen.getByTestId('action-area-title')).toHaveTextContent(
        'Custom Form Name',
      );
    });
  });

  describe('Translation Integration', () => {
    it('should use translation keys for button texts', () => {
      render(
        <ObservationFormsContainer {...defaultProps} viewingForm={mockForm} />,
      );

      expect(
        screen.getByText('translated_OBSERVATION_FORM_SAVE_BUTTON'),
      ).toBeInTheDocument();
      expect(
        screen.getByText('translated_OBSERVATION_FORM_DISCARD_BUTTON'),
      ).toBeInTheDocument();
      expect(
        screen.getByText('translated_OBSERVATION_FORM_BACK_BUTTON'),
      ).toBeInTheDocument();
    });
  });
});
