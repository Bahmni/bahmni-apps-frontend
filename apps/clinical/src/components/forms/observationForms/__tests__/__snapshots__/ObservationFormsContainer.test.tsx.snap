// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObservationFormsContainer Rendering and Structure should match the snapshot when not viewing a form 1`] = `<div />`;

exports[`ObservationFormsContainer Rendering and Structure should match the snapshot when viewing a form 1`] = `
<div>
  <div
    class="formViewActionArea"
    data-testid="action-area"
  >
    <div
      data-testid="action-area-title"
    >
      <div
        class="formTitleContainer"
      >
        <span>
          Test Form
        </span>
        <div
          class="pinIconContainer unpinned"
          title="Pin form"
        >
          <div
            data-icon-name="fa-thumbtack"
            data-size="SM"
            data-testid="icon-pin-icon"
          >
            Icon
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="action-area-content"
    >
      <div
        class="formView"
      >
        <div
          class="formContent"
        />
      </div>
    </div>
    <div
      data-testid="action-area-buttons"
    >
      <button
        data-testid="primary-button"
      >
        translated_OBSERVATION_FORM_SAVE_BUTTON
      </button>
      <button
        data-testid="secondary-button"
      >
        translated_OBSERVATION_FORM_DISCARD_BUTTON
      </button>
      <button
        data-testid="tertiary-button"
      >
        translated_OBSERVATION_FORM_BACK_BUTTON
      </button>
    </div>
  </div>
</div>
`;
